pkg/restic.go

```diff
@@ -0,0 +1,154 @@
+package pkg
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+
+	datamoverv1alpha1 "github.com/konveyor/volume-snapshot-mover/api/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Restic secret data keys
+const (
+	// AWS vars
+	AWSAccessKey     = "AWS_ACCESS_KEY_ID"
+	AWSSecretKey     = "AWS_SECRET_ACCESS_KEY"
+	AWSDefaultRegion = "AWS_DEFAULT_REGION"
+
+	// TODO: GCP and Azure
+
+	// Restic repo vars
+	ResticPassword   = "RESTIC_PASSWORD"
+	ResticRepository = "RESTIC_REPOSITORY"
+
+	// Datamover annotation keys
+	DatamoverResticRepository = "datamover.io/restic-repository"
+	DatamoverSourcePVCName    = "datamover.io/source-pvc-name"
+	DatamoverSourcePVCSize    = "datamover.io/source-pvc-size"
+)
+
+// Restic secret vars to create new secrets
+var (
+	AWSAccessValue        []byte
+	AWSSecretValue        []byte
+	AWSDefaultRegionValue []byte
+
+	// TODO: GCP and Azure
+
+	ResticPasswordValue []byte
+	ResticRepoValue     string
+)
+
+const (
+	resticSecretName = "restic-secret"
+	VSBLabel         = "datamover.oadp.openshift.io/vsb"
```

Comment:
could we use these labels that are in common.go?

Reply:
I tried to. It created a cyclic dependency and go compiler did not allow it..

Reply:
Ah gotcha